pandas/_libs/tslibs/np_datetime.pxd

```diff
@@ -120,3 +120,9 @@ cdef int64_t convert_reso(
     NPY_DATETIMEUNIT to_reso,
     bint round_ok,
 ) except? -1
+
+cdef extern from "src/datetime/np_datetime_strings.h":
+    cdef enum Exact:
+        PARTIAL_MATCH
+        EXACT_MATCH
+        NO_MATCH
```

Comment:
Does `NoMatch` really mean that the format is inferred?

Reply:
You're right, I've renamed to `INFER_FORMAT`, thanks!